molgenis-app-manager/src/test/java/org/molgenis/app/manager/controller/AppControllerTest.java

```diff
@@ -91,40 +111,59 @@ public void beforeMethod()
 		when(app.getAppConfig()).thenReturn("{'config': 'test'}");
 		ClassLoader classLoader = getClass().getClassLoader();
 		File testFile = new File(classLoader.getResource("test-resources/js/test.js").getFile());
-		String absoluteTestFileResourcePath = testFile.getPath().replace("js/test.js", "");;
+		String absoluteTestFileResourcePath = testFile.getPath().replace("js/test.js", "");
+		;
 		when(app.getResourceFolder()).thenReturn(absoluteTestFileResourcePath);
 
 		appResponse = AppResponse.create(app);
 		when(appManagerService.getAppByUri("uri")).thenReturn(appResponse);
 
-		AppController controller = new AppController(appManagerService, appSettings, menuReaderService);
-		mockMvc = MockMvcBuilders.standaloneSetup(controller)
-								 .setControllerAdvice(new GlobalControllerExceptionHandler(),
-										 new FallbackExceptionHandler(), new SpringExceptionHandler())
+		mockMvc = MockMvcBuilders.standaloneSetup(appController)
+								 .setControllerAdvice(globalControllerExceptionHandler, fallbackExceptionHandler,
+										 springExceptionHandler)
 								 .setLocaleResolver(localeResolver)
 								 .build();
 	}
 
+	@AfterMethod
+	public void afterMethod()
+	{
+		PluginIdentity pluginIdentity = new PluginIdentity("uri");
+		when(userPermissionEvaluator.hasPermission(pluginIdentity, VIEW_PLUGIN)).thenReturn(false);
+	}
+
 	@Test
 	public void testServeApp() throws Exception
 	{
+		PluginIdentity pluginIdentity = new PluginIdentity("uri");
+		when(userPermissionEvaluator.hasPermission(pluginIdentity, VIEW_PLUGIN)).thenReturn(true);
 		mockMvc.perform(get(AppController.URI + "/uri/"))
 			   .andExpect(status().isOk())
 			   .andExpect(model().attribute("app", appResponse))
 			   .andExpect(model().attribute("baseUrl", "/test/path"))
 			   .andExpect(view().name("view-app"));
 	}
 
+	@Test
+	public void testServeAppNoPermissions() throws Exception
+	{
+		mockMvc.perform(get(AppController.URI + "/uri/")).andExpect(status().isUnauthorized());
+	}
+
 	@Test
 	public void testServeAppRedirectToApp() throws Exception
 	{
-		mockMvc.perform(get(AppController.URI + "/uri"))
-			   .andExpect(status().is3xxRedirection());
+		PluginIdentity pluginIdentity = new PluginIdentity("uri");
+		when(userPermissionEvaluator.hasPermission(pluginIdentity, VIEW_PLUGIN)).thenReturn(true);
+		mockMvc.perform(get(AppController.URI + "/uri")).andExpect(status().is3xxRedirection());
```

Comment:
Could also check the location you're redirected to. (But existing test code already doesn't do this).